### PR TITLE
ACTIVITI-1991 Task is not updated in Query

### PR DESCRIPTION
**Pull request by lucianoprea**
-added listener delegate for TaskUpdate event
-created a test for this listener

refs [1991](https://github.com/Activiti/Activiti/issues/1991)

### CODE REVIEW

File: activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToAPITaskUpdatedEventConverter.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    private final APITaskConverter taskConverter;
+
+    public ToAPITaskUpdatedEventConverter(APITaskConverter taskConverter) {
+        this.taskConverter = taskConverter;
+    }
+
+    @Override
+    public Optional<TaskUpdatedEvent> from(ActivitiEntityEvent internalEvent) {
+        return Optional.of(new TaskUpdatedEventImpl(taskConverter.from((org.activiti.engine.task.Task) internalEvent.getEntity())));
```
Comment (erdemedeiros) [inline activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToAPITaskUpdatedEventConverter.java]: Here you need to check before if it's a Task update or an update of another entity. As you are using `ActivitiEventType.ENTITY_UPDATED` to register the listener it's listening for all the entity updates, not only for tasks. See https://github.com/Activiti/Activiti/blob/develop/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToTaskSuspendedConverter.java#L37.

File: activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@ContextConfiguration
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
Comment (erdemedeiros) [inline activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java]: We definitely need to find a better way of doing this. I don't like this FixMethodOrder, but for now that's the only way we got it working with `@WithUserDetails` annotation.

File: activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                                                         .withPriority(50)
+                                                         .withAssignee(authenticatedUserId)
+                                                         .build());
+
+        assertThat(RuntimeTestConfiguration.createdTasks).contains(standaloneTask.getId());
+
+        Page<Task> tasks = taskRuntime.tasks(Pageable.of(0,
+                                                         50));
+
+        assertThat(tasks.getTotalItems()).isEqualTo(1);
```
Comment (erdemedeiros) [inline activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java]: IMO this assertion is not necessary.

File: activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        // claim
+        taskRuntime.claim(TaskPayloadBuilder.claim().withTaskId(standaloneTask.getId()).build());
+
+        // update
+        final Task updatedTask = taskRuntime.update(updateTaskPayload);
+        tasks = taskRuntime.tasks(Pageable.of(0,
+                                              50));
+
+        assertThat(RuntimeTestConfiguration.updatedTasks).contains(updatedTask.getId());
+        assertThat(tasks.getTotalItems()).isEqualTo(2);
```
Comment (erdemedeiros) [inline activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java]: Same here, I don't think you need to check the total elements.

File: activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java (hunk 2 above)
Comment (lucianoprea) [reply activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java]: I don't like this either but this way are configured all tests for tasks due to the user scenarios I believe.

Comment (erdemedeiros) [reply activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java]: yeah, unfortunately we'll need to live with that until we figure out how to switch of user when using `@WithUserDetails`.

File: activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToAPITaskUpdatedEventConverter.java (hunk 1 above)
Comment (lucianoprea) [reply activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToAPITaskUpdatedEventConverter.java]: I missed that, makes total sense to have this check.

File: activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java (hunk 4 above)
Comment (lucianoprea) [reply activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/tasks/TaskRuntimeUpdateTaskTest.java]: I removed it.

**Comment by lucianoprea** [comment]
@erdemedeiros I did the changes requests but github is not showing the latests changes. I've rebased my branch on latest develop

**Pull request merged**